draftjs_exporter/dom.py

```diff
@@ -3,155 +3,125 @@
 import inspect
 import re
 
-from bs4 import BeautifulSoup
+from draftjs_exporter.dom_engine import DOM_HTML5LIB, DOM_LXML
+from draftjs_exporter.error import ConfigException
 
 # Python 2/3 unicode compatibility hack.
 # See http://stackoverflow.com/questions/6812031/how-to-make-unicode-string-with-python3
 try:
     UNICODE_EXISTS = bool(type(unicode))
 except NameError:
-    unicode = lambda s: str(s)
+    def unicode(s):
+        return str(s)
 
 # https://gist.github.com/yahyaKacem/8170675
 _first_cap_re = re.compile(r'(.)([A-Z][a-z]+)')
 _all_cap_re = re.compile('([a-z0-9])([A-Z])')
 
 
-def Soup(raw_str):
-    """
-    Wrapper around BeautifulSoup to keep the code DRY.
-    """
-    return BeautifulSoup(raw_str, 'html5lib')
-
-
-# Cache empty soup so we can create tags in isolation without the performance overhead.
-soup = Soup('')
-
-
-def create_tag(type_, attributes=None):
+class DOM(object):
     """
-    Wrapper around our HTML building library to facilitate changes.
+    Component building API, abstracting the DOM implementation.
     """
-    if attributes is None:
-        attributes = {}
-
-    return soup.new_tag(type_, **attributes)
-
 
-def create_document_fragment():
-        return create_tag('fragment')
+    HTML5LIB = 'html5lib'
+    LXML = 'lxml'
 
+    dom = DOM_HTML5LIB
 
-def get_text_content(elt):
-        return elt.string
-
-
-def set_text_content(elt, text):
-        elt.append(text)
+    @staticmethod
+    def camel_to_dash(camel_cased_str):
+        sub2 = _first_cap_re.sub(r'\1-\2', camel_cased_str)
+        dashed_case_str = _all_cap_re.sub(r'\1-\2', sub2).lower()
+        return dashed_case_str.replace('--', '-')
 
+    @classmethod
+    def use(cls, engine=DOM_HTML5LIB):
+        """
+        Choose which DOM implementation to use.
+        """
+        if engine:
+            if inspect.isclass(engine):
+                cls.dom = engine
+            elif engine.lower() == cls.HTML5LIB:
+                cls.dom = DOM_HTML5LIB
+            elif engine.lower() == cls.LXML:
+                cls.dom = DOM_LXML
+            else:
```

Comment:
Ah good point, I didn't think of not tying this to Django. Yeah the function doesn't seem too complicated, although it uses six which we will need to add as a dependency.

Agreed, this doesn't have to be tackled in this PR and can be considered an enhancement.

That being said, with the proposed change below of splitting the engines into different files, it would be really handy. The alternative to that is to import the module within the `if` statement like so:
```python
if engine.lower() == cls.HTML5LIB:
    from .engines.html5lib import DOM_HTML5LIB
    cls.dom = DOM_HTML5LIB
elif engine.lower() == cls.LXML:
    from .engines.lxml import DOM_LXML
    cls.dom = DOM_LXML
```